File: src/Deviser.Core/Deviser.Core.Data/Entities/ContentType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Deviser.Core.Data.Entities
{
    public class ContentType
    {
        public Guid Id { get; set; }
        public string Name { get; set; } //old property - Type        
        public string Label { get; set; }
        public string IconImage { get; set; }
        public string IconClass { get; set; }
        public int SortOrder { get; set; }
        public Guid ContentDataTypeId { get; set; }
        public ContentDataType ContentDataType { get; set; }
        public ICollection<ContentTypeProperty> ContentTypeProperties { get; set; }
        public ICollection<PageContent> PageContents { get; set; }
        public bool IsActive { get; set; }
        public DateTime? CreatedDate { get; set; }
        public DateTime? LastModifiedDate { get; set; }
    }
}

File: src/Deviser.Core/Deviser.NgLib/deviser-workspace/projects/deviser-shared/src/lib/domain-types/content-type.ts
import { Property } from './property';
import { ContentTypeField } from './content-type-field';

export interface ContentType {
    id: string;
    name: string;
    label: string;
    isList: boolean;
    iconImage: string;
    iconClass: string;
    sortOrder: number;
    properties: Property[];
    contentTypeFields: ContentTypeField[];
    isActive: boolean;
    isActiveBadgeClass: string;
    createdDate: Date | string | null;
    lastModifiedDate: Date | string | null;
}